### DISCUSSION TITLE
Create a CommonRepository according to my CommonEntity - type error

### DISCUSSION THREAD

**Discussion by billomore** [mikro-orm/mikro-orm#5002] [Q&A] [answered]
Hi, 

Question. All my entities extend a `CommonEntity`:
```typescript
export class CommonEntity {

    @PrimaryKey({ columnType: 'uuid', defaultRaw: `uuid_generate_v4()` })
    id = uuidv4();
}
```
So I was trying to create a `CommonRepo`, that will do some common stuff, e.g. like this:
```typescript
export class CommonRepo<EntityType extends CommonEntity> extends EntityRepository<EntityType> {

    async getById(id: string) {
        return this.findOne( { id } );
    }
}
```
But I'm getting a type error on the `{ id }`: 
`Argument of type '{ id: string; }' is not assignable to parameter of type 'FilterQuery<EntityType>'.
  Object literal may only specify known properties, and 'id' does not exist in type 'NonNullable<EntityProps<EntityType> & OperatorMap<EntityType>> | FilterQuery<EntityType>`

I thought I told it to extend my `CommonEntity`; what did I miss?

Having great time with MikroOrm btw, thanks for the project!

**Comment by B4nan** [comment, marked as answer]
This is a TS limitation, you need a type assertion here, TS won't respect the extends clause, unfortunately.

https://stackoverflow.com/questions/65490587/generics-doesnt-seem-to-fit-well-with-queryfilter/65523491#65523491

**Reply by billomore** [reply]
ouch :(
so it should go like this?
```typescript
async getById(id: string) {
    return this.findOne( { id } as FilterQuery<EntityType> );
}
```

**Reply by B4nan** [reply]
Yes. Note that you can do `em.findOne(Type, id)` right ahead.
